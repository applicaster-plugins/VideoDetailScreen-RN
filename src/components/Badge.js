import React from 'react';
import PropTypes from 'prop-types';
import Text from '@applicaster/london-rn-components/src/SafeText';

const Badge = ({ style, labelText }) => <Text style={style}>{labelText}</Text>;

Badge.propTypes = {
  style: PropTypes.any,
  labelText: PropTypes.string
};

export default Badge;
