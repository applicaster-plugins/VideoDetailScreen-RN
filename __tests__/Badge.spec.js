import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Badge from '../src/components/Badge';

jest.mock('../src/utils/zappConfig');

const style = {
  position: 'absolute',
  zIndex: 0,
  right: 0,
  bottom: 0,
  padding: 7,
  fontFamily: 'Apertura-Medium',
  color: '#FFFFFF',
  fontSize: 11,
  backgroundColor: 'rgba(231, 199, 108, .7)'
};

it('Badge renders correctly', () => {
  const tree = renderer
    .create(<Badge style={style} labelText="Badge Text" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
