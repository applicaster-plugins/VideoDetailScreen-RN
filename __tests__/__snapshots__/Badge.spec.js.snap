// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "backgroundColor": "rgba(231, 199, 108, .7)",
      "bottom": 0,
      "color": "#FFFFFF",
      "fontFamily": "Apertura-Medium",
      "fontSize": 11,
      "padding": 7,
      "position": "absolute",
      "right": 0,
      "zIndex": 0,
    }
  }
>
  Badge Text
</Text>
`;
